### PR TITLE
DM-32081: fix a doc string to show modify_environ from utils, not salobj. Modernize the Jenkinsfile.

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: doc/version_history.rst [hunk 1]
```diff
@@ -6,6 +6,13 @@
 Version History
 ###############
 
+v1.1.1
+------
+
+Fix a doc string to show using modify_environment from utils, not salobj.
```
Comment (tribeiro) [inline doc/version_history.rst:12]: uhm.. I don't see this change in the git diff. Am I missing something?

File: doc/version_history.rst [hunk 2]
```diff
@@ -6,6 +6,13 @@
...
+
+Refine Jenkinsfile.
```
Comment (tribeiro) [inline doc/version_history.rst:14]: In mtdometrajectory you called it "Modernize Jenkinsfile." maybe follow the same pattern?

**Review by tribeiro — APPROVED** [review]
Looks good but the version history mentions a change I could not find. Maybe take a look at it before merging?

File: doc/version_history.rst (hunk 1 above)
Comment (r-owen) [reply doc/version_history.rst:12]: You are right. I had already committed that to develop without updating the version history.

File: doc/version_history.rst (hunk 2 above)
Comment (r-owen) [reply doc/version_history.rst:14]: FYI: the reason I worded it differently is that the Jenkinsfile already followed the old template (which is more modern than most of my packages). But it's a fair request and I'll do it.

**Pull request merged**